packages/payment/src/loaders/providers.ts

```diff
@@ -1,24 +1,18 @@
 import { moduleProviderLoader } from "@medusajs/modules-sdk"
-
 import { LoaderOptions, ModuleProvider, ModulesSdkTypes } from "@medusajs/types"
-import { Lifetime, asFunction } from "awilix"
+import { Lifetime, asFunction, asValue } from "awilix"
 
 const registrationFn = async (klass, container, pluginOptions) => {
-  container.register({
-    [`payment_provider_${klass.prototype}`]: asFunction(
-      (cradle) => new klass(cradle, pluginOptions),
-      {
+  Object.entries(pluginOptions.config).map(([k, v]) => {
+    const key = `pp_${klass.PROVIDER}_${k}`
+    container.register({
+      [key]: asFunction((cradle) => new klass(cradle, pluginOptions), {
         lifetime: klass.LIFE_TIME || Lifetime.SINGLETON,
-      }
-    ),
-  })
-
-  container.registerAdd(
-    "payment_providers",
-    asFunction((cradle) => new klass(cradle, pluginOptions), {
-      lifetime: klass.LIFE_TIME || Lifetime.SINGLETON,
+      }).inject(() => ({ config: v })),
```

Comment:
todo: since we already have the config at hand, I would probably pass it in the plugin options associated to a specifi key such as `currentProviderConfig` or similar. The inject can lead to conflict with other registration that module dependencies could include. wdyt?